Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py

```diff
@@ -90,20 +91,57 @@ def editable(cls, branch, repository=None):
         return True
 
     @classmethod
-    def branch_point(cls, repository, limit=None):
-        cnt = 0
-        commit = repository.commit(include_log=False, include_identifier=False)
-        while cls.editable(commit.branch, repository=repository):
-            cnt += 1
-            if limit and cnt > limit:
-                return None
-            commit = repository.find(argument='HEAD~{}'.format(cnt), include_log=False, include_identifier=False)
-            if cnt > 1 or commit.branch != repository.branch or cls.editable(commit.branch, repository=repository):
-                log.info('    Found {}...'.format(string_utils.pluralize(cnt, 'commit')))
-            else:
-                log.info('    No commits on editable branch')
+    def branch_point(cls, repository):
+        branches = repository.branches_for(remote=None)
+        production_branches = [
+            'remotes/{}/{}'.format(remote, branch)
+            for remote in repository.source_remotes()
+            for branch in branches[remote] if not repository.dev_branches.match(branch)
+        ]
```

Comment:
This feels like something we could just put on `Repository` rather than having it inline here?

Reply:
This is the only place we need this particular list, at the moment. I'm not against moving this to `Repository` in the future, but for the moment, I don't. see where else we'd need to use this.

Reply:
Yeah, no strong opinion here.